docs/04-docker-setup-postgres-connect-with-r.md

```diff
@@ -91,21 +94,35 @@ system2("docker", "ps", stdout = TRUE, stderr = TRUE)
 ```
 
 ```
-## [1] "CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES"    
-## [2] "fdbefcf484f1        postgres:10         \"docker-entrypoint.s…\"   37 seconds ago      Up 11 seconds       0.0.0.0:5432->5432/tcp   sql-pet"
+## [1] "CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                  PORTS                    NAMES"   
+## [2] "596ad8bf2df8        postgres:10         \"docker-entrypoint.s…\"   1 second ago        Up Less than a second   0.0.0.0:5432->5432/tcp   cattle"
 ```
 ## Connect, read and write to Postgres from R
 
-Create a connection to Postgres after waiting 3 seconds so that Docker has time to do its thing.
+Create a connection to Postgres waiting up to 10 seconds for the database to come up in Docker.
 
-```r
-Sys.sleep(3)
 
-con <- DBI::dbConnect(RPostgres::Postgres(),
-                      host = "localhost",
-                      port = "5432",
-                      user = "postgres",
-                      password = "postgres")
+```r
+wait_for_postgres <- function(seconds_to_test){
+  for (i in 1:seconds_to_test) {
+    db_ready <- DBI::dbCanConnect(RPostgres::Postgres(),
+                        host = "localhost",
+                        port = "5432",
+                        user = "postgres",
+                        password = "postgres")
+    if ( !db_ready ) {Sys.sleep(1)} 
```

Comment:
If `!db_ready` is TRUE, you need to look at `attr(db_ready, "result")` before going into the loop. It's a text string; if it contains "FATAL" the database is up but your connection parameters aren't valid. Trying again 30 more times won't change that!

Reply:
addressed in an updated commit.